Add more changelog info to PR template (#843)
PlasmaPy/PlasmaPy: .github/PULL_REQUEST_TEMPLATE.md

Thank you for contributing to PlasmaPy! Here's a bunch of pointers to
make things easier for all of us:

> * If this PR will solve an issue tracked by GitHub, please add
>   "Closes #42." for example issue 42 to automatically close the issue once the
>   pull request is merged.  If your PR will not completely solve the issue,
>   please still reference the issue.

Reviewer: ```suggestion
* If this PR will solve an issue tracked by GitHub, then please add
  "Closes #42" so the issue automatically closes once this pull request
  is merged.  In this example, issue #42 would have been closed.
  If your PR will not completely solve the issue, then please still reference
  still reference the issue like "issue #42".  (For more info see [GitHub's primer
  on linking issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).)
```

* Remember to add some description of your changes in this text box.

* If your pull request is not yet ready for review - either because
  you're just looking for feedback on a change or because you're not
  perfectly satisfied with your change - submit it as a draft pull request.
  Remember to change its' status once it's ready.

* If this is your first contribution, please add your name to the author
  list in `docs/about/credits.rst`.

* Feel free to chat with other developers on our Matrix channel at:
   https://riot.im/app/#/room/#plasmapy:openastronomy.org

> * We have a developer's guide, where some answers to your questions may
>   possibly be found, at

Reviewer: ```suggestion
* We have a developer's guide to help answer some of your questions. 
```

-->

- [ ] I have added a changelog entry for this pull request.

<!--

> In short: changelog entries are short descriptions of your PR's changes
> in `.rst` files in the `changelog` directory, named like
> `<PULL REQUEST>.<TYPE>.rst`, where `<PULL
> REQUEST>` is a pull request number, and `<TYPE>` is one of:

Reviewer: ```suggestion
In short: A changelog entry is a short description of your PR's changes.
Each entry is written in a `<PULL REQUEST>.<TYPE>.rst` file and stored in 
the `changelog` directory,  where `<PULL REQUEST>` is a pull request 
number and `<TYPE>` is one of:
```

* `breaking`: A change which requires users to change code and is not backwards compatible. (Not to be used for removal of deprecated features.)
* `feature`: New user facing features and any new behavior.
* `bugfix`: Fixes a reported bug.
* `doc`: Documentation addition or improvement, like rewording an entire session or adding missing docs.
* `removal`: Feature deprecation and/or feature removal.
* `trivial`: A change which has no user facing effect or is tiny change.

> You get the PR number when you submit it, so this part needs a follow-up
> commit.

Reviewer: ```suggestion
A PR number is generated after you successfully submit a new PR, so the changelog
file can only be added after you open the PR.```

For more information, see:
https://github.com/PlasmaPy/PlasmaPy/blob/master/changelog/README.rst

--->

- [ ] If adding new functionality, I have added tests and
      docstrings.

<!--
(Tests pop up at the bottom, in the checks box.)
-->

- [ ] I have fixed any newly failing tests.

<!--

> (if you're unsure why they're failing, ask!)

Reviewer: ```suggestion
(If you're unsure why they're failing, ask!)
```
